CONTRIBUTING.md

```diff
@@ -0,0 +1,34 @@
+# Contribution Guide <!-- omit in toc -->
+
+Thank you for investing your time in contributing to our project! Please read our [Code of Conduct](./CODE_OF_CONDUCT.md) to keep our community approachable and respectable.
+
+This guide provides an overview of the contribution workflow from opening an issue, creating a PR, reviewing, and merging the PR.
+
+## Contributions
+
+When contributing to this repoository, please first discuss any changes you wish to make via an issue, email, or any other method with the owners of this repository before making a change.
+
+To make sure you understand the general project concepts, please read the project [README](README.md). 
+
+## Creating a new issue
+
+If you spot a problem with anything with the docs or with the library itself, [search the existing issues](https://github.com/Apollo-Protocol/hqdm-lib/issues). If a related issue doesn't exist, open a [new issue](https://github.com/Apollo-Protocol/hqdm-lib/issues/new).
+
+## Searching Issues
+
+Scan through our [existing issues](https://github.com/Apollo-Protocol/hqdm-lib/issues) to find one that interests you. You can narrow down the search using `labels` as filters. If you find an issue to work on, please start a discussion to make sure the scope is clear and once you start development assign yourself to the issue. Once happy, open a PR with a fix or enhancement.
+
+## Issue Workflow
+* Choose an issue you wish to resolve. Where possible a pull request should correlate to a single issue.
+* Assign yourself to the issue
+* Fork the repository to your account.
+* Create a branch using the pattern [type]/[issue number]-[issue title]. Branching follows [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow).
```

Comment:
If we're telling people to fork the repo I'm not sure we need to tell them how to name their branches?

Reply:
I don't think so. That can be discussed more privately with anyone who gets given write access to the main repo.

Reply:
Sounds sensible. Updated on next commit.